include/dlaf/communication/sync/broadcast.h

```diff
@@ -48,5 +52,42 @@ void receive_from(const int broadcaster_rank, Communicator& communicator, DataOu
 }
 }
 }
+
+template <class T>
+void send_tile(hpx::threads::executors::pool_executor ex,
+               common::Pipeline<comm::CommunicatorGrid>& task_chain, Coord rc_comm,
+               hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile) {
+  using ConstTile_t = matrix::Tile<const T, Device::CPU>;
+  using PromiseComm_t = common::PromiseGuard<comm::CommunicatorGrid>;
+
+  auto send_bcast_f = hpx::util::annotated_function(
+      [rc_comm](hpx::shared_future<ConstTile_t> ftile, hpx::future<PromiseComm_t> fpcomm) {
+        PromiseComm_t pcomm = fpcomm.get();
+        comm::sync::broadcast::send(pcomm.ref().subCommunicator(rc_comm), ftile.get());
+      },
+      "send_tile");
+  hpx::dataflow(ex, std::move(send_bcast_f), tile, task_chain());
+}
+
+template <class T>
+hpx::future<matrix::Tile<const T, Device::CPU>> recv_tile(
+    hpx::threads::executors::pool_executor ex, common::Pipeline<comm::CommunicatorGrid>& mpi_task_chain,
+    Coord rc_comm, TileElementSize tile_size, int rank) {
+  using ConstTile_t = matrix::Tile<const T, Device::CPU>;
+  using PromiseComm_t = common::PromiseGuard<comm::CommunicatorGrid>;
+  using MemView_t = memory::MemoryView<T, Device::CPU>;
+  using Tile_t = matrix::Tile<T, Device::CPU>;
+
+  auto recv_bcast_f = hpx::util::annotated_function(
+      [rank, tile_size, rc_comm](hpx::future<PromiseComm_t> fpcomm) -> ConstTile_t {
+        PromiseComm_t pcomm = fpcomm.get();
+        MemView_t mem_view(tile_size.linear_size());
+        Tile_t tile(tile_size, std::move(mem_view), tile_size.rows());
+        comm::sync::broadcast::receive_from(rank, pcomm.ref().subCommunicator(rc_comm), tile);
+        return std::move(tile);
```

Comment:
I think the move here is unnecessary and potentially harmful, at least on the compilers you are likely to use. It's not compulsory to do RVO here (specifically NRVO), meaning it might be required in some cases, but both GCC and clang do it here. (See NRVO here: https://en.cppreference.com/w/cpp/language/copy_elision).

Reply:
We need it there otherwise we get a compiler error:

```
/home/teonnik/code/dlaf/include/dlaf/communication/sync/broadcast.h:88:16: error: call to deleted constructor of 'ConstTile_t' (aka 'Tile<const std::complex<double>, Device::CPU>')
        return tile;
```

The reason is that there is actually a conversion from Tile_t to ConstTile_t. The latter expects an rvalue of the former.

Reply:
Thanks for the explanation! I missed the different types.